### ISSUE TITLE
Version 4.0.1 - manual refresh is need to show dashboard

### ISSUE THREAD

**Issue by Slaheddine** [apache/superset#28833]
### Bug description

After migrating to the app version 4.0.1 and Helm Chart 0.12.11, our dashboards are not showing any more and stay in loading state without showing the result. We are obliged to manually refresh the page to have the results.

### How to reproduce the bug

using docker images : 
- superset 4.0.1
- superset 4.0.1-dockerize
- superset 4.0.1-websocket
and Helm Chart : 0.12.11

### Screenshots/recordings

[
<img width="1219" alt="Screenshot 2024-06-03 at 14 32 13" src="https://github.com/apache/superset/assets/7395576/c15b4805-1433-48fb-b34f-0e7c7f1dffe5">
](url)

### Superset version

4.0.1

### Python version

3.9

### Node version

Not applicable

### Browser

Not applicable

### Additional context

the only logs we have : 

│ superset [2024-06-03 11:31:49,932: WARNING/MainProcess] /usr/local/lib/python3.9/site-packages/celery/worker/consumer/consumer.py:507: CP ││ endingDeprecationWarning: The broker_connection_retry configuration setting will no longer determine                                      │

│ superset whether broker connection retries are made during startup in Celery 6.0 and above.                                               ││ superset If you wish to retain the existing behavior for retrying connections on startup,                                                 │

│ superset you should set bro

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by rusackas** [comment]
Nope, no updates, and seemingly nobody else experiencing this. Are you getting any errors or showing any additional logs at all?

**Comment by Slaheddine** [comment]
Changing the GLOBAL_ASYNC_QUERIES_TRANSPORT from "ws" to "polling" and downgrading to version 2.1.1. I think there is a problem with the websocket mode.

**Comment by talfour** [comment]
Having same issue with websockets using docker. 
Superset version = 4.0.1 Superset websocket version = 4.0.1 Python version =3.9

**Comment by rusackas** [comment]
Oooh, interesting. I'm not sure who's using the websocket mode in day-to-day ops presently. Not sure if @craig-rueda @villebro or @michael-s-molina have tried it out lately.

**Comment by MalavikaN1** [comment]
We are also facing the same issue, using superset image version 3.1.1.
Any fixes for this?

**Comment by pasqualtroncone** [comment]
I'm facing somewhat the same problem after upgrading from 3.1.2 to 3.1.3. In my case, the dashboard will load the first time, but some charts do not load when filters are applied.

**Comment by rusackas** [comment]
@pasqualtroncone are you getting any errors in the JS console, or in the server logs?

**Comment by max-viafoura** [comment]
We are also seeing this after upgrading from 3.1.1 to 4.0.1
Using docker to deploy it on EKS. Not seeing errors on front end but seeing a lot of `Received unexpected request: GET /"` on websocket pod

**Comment by max-viafoura** [comment]
Update - the above was issue with infrastructure on our end - restarting the pod resolved the messaging but did not solve a problem. Right now we can select filters and dashboard loads initially but once anyone apply filter - we see loading icon but no errors in web-dev console or on server. And `data` requests are getting `pending` result with nothing else happening:
![image](https://github.com/apache/superset/assets/97770427/061a9485-3dfc-4fcf-bec3-83915d42aa13)

**Comment by pasqualtroncone** [comment]
> @pasqualtroncone are you getting any errors in the JS console, or in the server logs?

@rusackas, nothing on the JS console or server logs, but it seems that my case is related to Firefox. It takes way too much time compared to Chrome to give back some results. At first, I thought it was an endless loop, but then the browser advised closing the tab, so the problem was clear. I solved my issue by not using Firefox. I think the cause of this comes from Pivot Table component.
For the same dashboard, Chrome takes 3 seconds, while Firefox takes over 40 seconds.

![image](https://github.com/user-attachments/assets/24e4f5f0-9222-4091-92b6-3db8a318e432)

**Comment by rusackas** [comment]
Hm, interesting! @betodealmeida have you run into this? You're the only avid Firefox user I know :D

**Comment by sadpandajoe** [comment]
> I am using websockets and I see this issue only when SLACK_ENABLE_AVATARS is enabled..

@mistercrunch thoughts on this as the creator of: https://github.com/apache/superset/pull/27849

**Comment by mistercrunch** [comment]
No direct/quick intuition on how `SLACK_ENABLE_AVATARS` relates to websocket/firefox.

Have we identified this is specific to GLOBAL_ASYNC_QUERIES_TRANSPORT + FIREFOX?

I'd qualify `GLOBAL_ASYNC_QUERIES_TRANSPORT` as bleeding edge currently as I think few large install based use it in production but that's just a guess, we don't have hard data on it. I can say for sure we don't use GLOBAL_ASYNC_QUERIES_TRANSPORT at Preset currently but intend to do so in the nearish future as the related SIP and work takes place.

**Comment by mistercrunch** [comment]
I do not know the details about the test coverage of `GLOBAL_ASYNC_QUERIES_TRANSPORT`, but hear @villebro use it in his production cluster. Preset currently uses `polling` in its installation, and `SLACK_ENABLE_AVATARS` is getting rolled out progressively through a feature flag (we use split.io behind the scene to support dynamic feature flag configuration and progressive rollouts). It may be a good idea to label `GLOBAL_ASYNC_QUERIES_TRANSPORT` as beta or alpha.

**Comment by jorgehn52** [comment]
Hello all.

I am facing the same issue for charts loading forever with version 4.0.1 and 4.0.2, on Superset installed with helm 0.12.11.

I already created another issue for the same in this [link](https://github.com/apache/superset/issues/30650)
Also noticing that some configs require several DB inside Redis but the helm chart is only creating:

REDIS_DB=1
REDIS_CELERY_DB=0

Is this a limitation?

**Comment by rusackas** [comment]
This has been quiet for quite some time. Glad it sounds like SLACK_ENABLE_AVATARS isn't the culprit, but is anyone able to confirm or deny if this is still happening in 4.1.2? Updated context would be appreciated.

**Comment by magnetic5355** [comment]
It's happening. Fresh install - thought I had configured something incorrectly. 

Using helm and Web socket

**Comment by rusackas** [comment]
Still being reported in July 2025 with Helm deployments using WebSockets. Common culprits are Gunicorn worker exhaustion or misconfigured WebSocket/async query settings. Ensure your deployment has sufficient Gunicorn workers and that `GLOBAL_ASYNC_QUERIES` is properly configured. If you have a specific stack trace or network timeline, please share it for further investigation.

**Comment by rusackas** [comment]
All reports trace to GLOBAL_ASYNC_QUERIES websocket transport on Helm/K8s deployments. IT seems switching transport to 'polling' or disabling GAQ resolves it. Closing.
